File: PHP/Day04.php
<?php

$string = file_get_contents('../inputs/Day04.txt');

if( empty($string) ){
	exit("Unable to load input file");
}

$fiveZerosString = '';
$sixZerosString = '';
$num = 0;
while($fiveZerosString === '' || $sixZerosString === ''){
	$hash = md5($string . $num);

	if( empty($fiveZerosString) && substr($hash, 0, 5) === '00000'){
		$fiveZerosString = $num;
	}
	
	if( empty($sixZerosString) && substr($hash, 0, 6) === '000000'){
		$sixZerosString = $num;
	}

	$num++;
}

print "
Part 1: {$fiveZerosString}
Part 2: {$sixZerosString}
";

File: Python/Day04.py
from hashlib import md5

string = open('../inputs/Day04.txt').read()

fiveZeros = ''
sixZeros = ''
num = 1

while fiveZeros == '' or sixZeros == '':
	hashed = md5(string + str(num)).hexdigest()

	if fiveZeros == '' and hashed[:5] == '00000':
		fiveZeros = num

	if sixZeros == '' and hashed[:6] == '000000':
		sixZeros = num

	num += 1

print "Part 1:", fiveZeros
print "Part 2:", sixZeros